controllers/user.controller.js

```diff
@@ -0,0 +1,157 @@
+const { User } = require("../models/user.model");
+
+var jwt = require("jsonwebtoken");
+require("dotenv").config();
+const JWT_KEY = process.env.JWT_KEY;
+var bcrypt = require("bcrypt");
+
+const { extend } = require("lodash");
+
+const addNewUser = async (req, res) => {
+   try {
+      const { username, email, password } = req.body;
+      const user = { userName: username, email: email, password: password };
+      const NewUser = new User(user);
+      const salt = await bcrypt.genSalt(10);
+      NewUser.password = await bcrypt.hash(NewUser.password, salt);
+      await NewUser.save();
+
+      NewUser.__v = undefined;
+      NewUser.password = undefined;
+
+      res.status(200).json({ message: "user added", NewUser });
+   } catch (error) {
+      console.log(error);
+      if (error.name === "MongoError" && error.code === 11000) {
+         if (error.keyPattern.userName) {
+            return res.status(422).json({
+               message: "user name is already taken",
+               errorMessage: error.message,
+            });
+         } else if (error.keyPattern.email) {
+            return res.status(422).json({
+               message: "account already exists with this email",
+               errorMessage: error.message,
+            });
+         }
+      }
+      res.status(500).json({
+         message: "could not add user",
+         errorMessage: error.message,
+      });
+   }
+};
+
+const loginUser = async (req, res) => {
+   try {
+      const { username, password } = req.body;
+
+      let user = await User.findOne({ userName: username });
+
+      if (user) {
+         const verifyPassword = await bcrypt.compare(password, user.password);
+         if (verifyPassword) {
+            const token = jwt.sign({ userId: user._id }, JWT_KEY, { expiresIn: "24h" });
+            user.__v = undefined;
+            user.password = undefined;
+            return res.status(200).json({ message: "user logged in", user, token });
+         }
+         return res.status(403).json({ message: "username and password did not match" });
+      }
+      res.status(404).json({ message: "user does not exist " });
+   } catch (error) {
+      res.status(500).json({
+         message: "cannot retrieve user",
+         errorMessage: error.message,
+      });
+   }
+};
+
+const resetOrUpdateUserPassword = async (req, res) => {
+   try {
+      const { email, password } = req.body;
```

Comment:
Taking `email` in reset password doesn't seem appropriate, because if let's say I know your `email`, I can just enter it and reset your password very easily!
For now, it's okay. You can change its logic later on!
Logic would be -> Take `email` in `forgotPassword` API and generate a token and send it to user email!
Then in `resetToken` API verify that token and give user permission to reset Password

Reply:
ha true, forgot the password not make much sense but I just let it that way for now.
Thanks for the suggestion, surely will think of this later